server/services/service.ts

```diff
@@ -0,0 +1,10 @@
+import type { Request } from 'express'
+
+export default class Service {
+  token: string
```

Comment:
I don't think this is thread safe. 

Services are singletons so can't store mutable request scoped data.

Concurrent requests from multiple users may modify this between calling `withTokenFromRequest ` and actually using the token, meaning one user's token can be used to make API calls on behalf of another.

Reply:
Yeah, you're quite right. I didn't really think about services behaving like a singleton across the whole lifecycle of the application. Back to the drawing board it is!